# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/tabbrowser.js
# Commit: 4b76024db21c
# Full Hash: 4b76024db21cff006a7424a35c412c6e0a153eb0
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-09-22 03:47:42
# Regressor Bug: 1655866
# File Overlap Count: 1
# Description:
#   Bug 1655866: Part 5 - Use native PermitUnload implementation from front-end code. r=nika,mconley
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D88318
# ==============================================================================

diff -r 55fb069319f1 -r 4b76024db21c browser/base/content/tabbrowser.js
--- a/browser/base/content/tabbrowser.js	Mon Sep 21 22:41:01 2020 +0000
+++ b/browser/base/content/tabbrowser.js	Mon Sep 21 22:41:04 2020 +0000
@@ -2307,9 +2307,7 @@
 
     _mayDiscardBrowser(aTab, aForceDiscard) {
       let browser = aTab.linkedBrowser;
-      let permitUnloadFlags = aForceDiscard
-        ? browser.dontPromptAndUnload
-        : browser.dontPromptAndDontUnload;
+      let action = aForceDiscard ? "unload" : "dontUnload";
 
       if (
         !aTab ||
@@ -2318,7 +2316,7 @@
         this._windowIsClosing ||
         !browser.isConnected ||
         !browser.isRemoteBrowser ||
-        !browser.permitUnload(permitUnloadFlags).permitUnload
+        !browser.permitUnload(action).permitUnload
       ) {
         return false;
       }
@@ -3386,7 +3384,7 @@
     _hasBeforeUnload(aTab) {
       let browser = aTab.linkedBrowser;
       if (browser.isRemoteBrowser && browser.frameLoader) {
-        return PermitUnloader.hasBeforeUnload(browser.frameLoader);
+        return browser.hasBeforeUnload;
       }
       return false;
     },